contrib/golang/filters/http/source/cgo.cc

```diff
@@ -0,0 +1,135 @@
+#include "contrib/golang/filters/http/source/golang_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Golang {
+
+//
+// These functions may be invoked in another go thread,
+// which means may introduce race between go thread and envoy thread.
+//
```

Comment:
I'm pretty sure to avoid problems here you are going to have to post to the dispatcher running the filter on the envoy thread, otherwise I don't see how this can work? Especially in the local reply function below and continuing, etc.. I think you should be using a dispatcher to post and checking the weak lock both before and after the post. This is done in other similar cases.

Reply:
Yes, it is very important here, as you said, by using a envoy's dispatcher to post to make the some process only be executed in the envoy thread. As below code
https://github.com/envoyproxy/envoy/blob/a2f78efdf32feb169568f448acbbe05caf96dd3a/contrib/golang/filters/http/source/golang_filter.cc#L579-L596

Reply:
OK I see you do it elsewhere. Then please update this comment to be more clear that you cover this in the filter.

Reply:
Done.